File: backend/src/main/java/at/ac/tuwien/sepm/groupphase/backend/service/TicketService.java
package at.ac.tuwien.sepm.groupphase.backend.service;

import at.ac.tuwien.sepm.groupphase.backend.entity.Ticket;

import java.util.List;

public interface TicketService {

    /**
     * Find all valid tickets of a performance.
     *
     * @param eventId       the id of the event in which to search for the performance containing the tickets
     * @param performanceId the id of the performance from which the tickets should be loaded
     * @return the tickets of the performance
     */
    List<Ticket> findByEventIdAndPerformanceId(Long eventId, Long performanceId);
}

File: frontend/src/app/services/ticket.service.ts
import {Injectable} from '@angular/core';
import {Observable} from 'rxjs';
import {HttpClient} from '@angular/common/http';
import {Globals} from '../global/globals';
import {Ticket} from '../dtos/ticket';

@Injectable({
  providedIn: 'root'
})
export class TicketService {

  private readonly eventBaseUri: string = `${this.globals.backendUri}/events`;

  constructor(private httpClient: HttpClient, private globals: Globals) {
  }

  /**
   * Loads all valid tickets or reservations for a performance
   *
   * @param eventId id of the event of which the performance is a part of
   * @param performanceId id of performance from which the tickets should be loaded
   */
  getTicketsByEventAndPerformance(eventId: number, performanceId: number): Observable<Ticket[]> {
    console.log(`Load performance details for performance ${performanceId} from event ${eventId}`);
    return this.httpClient.get<Ticket[]>(`${this.eventBaseUri}/${eventId}/performances/${performanceId}/tickets`);
  }

}
